hasktorch/test/Torch/Typed/NamedTensorSpec.hs

```diff
@@ -56,12 +65,32 @@ data YCoCg a = YCoCg
 
 data RGBA a = RGBA a a a a deriving (Show, Eq, Generic)
 
+instance {-# OVERLAPS #-} (Coercible (vec n a) (Vector n a), KnownNat n, D.TensorLike (ToNestedList a), NamedTensorLike a) => NamedTensorLike (vec n a) where
+  type ToNestedList (vec n a) = [ToNestedList a]
+  toNestedList v = map (toNestedList @a) (V.toList (coerce v :: Vector n a))
+  asNamedTensor v = fromUnnamed . UnsafeMkTensor . D.asTensor $ toNestedList v
+  fromNestedList v = coerce (fmap fromNestedList . fromJust . V.fromList $ v :: Vector n a)
+  fromNamedTensor =  fromNestedList . D.asValue . toDynamic
+
+instance HasTypes Float Float where
+  types_ = id
+
+instance {-# OVERLAPS #-} (Generic (g a), Default (g a), HasTypes (g a) a, KnownNat (ToNat g), D.TensorLike (ToNestedList a), NamedTensorLike a) => NamedTensorLike (g a) where
+  type ToNestedList (g a) = [ToNestedList a]
+  toNestedList v = map (toNestedList @a) (flattenValues (types @a) v)
+  asNamedTensor v = fromUnnamed . UnsafeMkTensor . D.asTensor $ toNestedList v
+  fromNestedList v = replaceValues (types @a) def (fmap fromNestedList v)
+  fromNamedTensor =  fromNestedList . D.asValue . toDynamic
+
```

Comment:
When putting this in Torch.Typed.Tensor, ghc requires a lot of memory.
I don't know the cause.